src/main/java/com/yahoo/bullet/dsl/BulletDSLConfig.java

```diff
@@ -156,6 +187,35 @@
                  .checkIf(Validator::isString)
                  .unless(Validator::isNull)
                  .orFail();
+
+        // BulletDeserializer validation
+        VALIDATOR.define(DESERIALIZER_CLASS_NAME);
+        VALIDATOR.define(DESERIALIZER_AVRO_SCHEMA_FILE)
+                 .checkIf(Validator::isString)
+                 .castTo(BulletDSLConfig::readFile)
+                 .unless(Validator::isNull)
```

Comment:
I think you also don't want to do this if you've already done the casting to a file. Otherwise, a validate after the fact will cause the check to fail.

Reply:
It casts to string

Reply:
Oh right...forgot the check is not for the `file:`